cli/src/diff_util.rs

```diff
@@ -37,7 +38,7 @@ use crate::ui::Ui;
 
 #[derive(clap::Args, Clone, Debug)]
 #[command(group(clap::ArgGroup::new("short-format").args(&["summary", "types"])))]
-#[command(group(clap::ArgGroup::new("long-format").args(&["git", "color_words", "tool"])))]
+#[command(group(clap::ArgGroup::new("long-format").args(&["git", "color_words", "tool", "stat"])))]
```

Comment:
I moved it to the "short-format" section but I suspect more changes would be needed to support combining options like this no? Even `summary` with `types` doesn't work:

```
$ jj diff --summary --types  --from=d039008d --to=78170b42
error: the argument '--summary' cannot be used with '--types'
```

Reply:
I'm going to leave this for now and maybe take a look at what it'd take to allow combining the options.

Reply:
> maybe take a look at what it'd take to allow combining the options.

I *think* it's just a matter of UX - do we want to allow `jj diff --git --color-words` or do we want to tell the user that they probably made a mistake? It might be okay to just drop the restriction.